src/views/common/Animation/AnimationStateReducer.tsx

```diff
@@ -137,36 +137,43 @@ const AnimationStateReducer = <T, >(s: AnimationState<T>, a: AnimationStateActio
                 console.warn('Attempt to set ms-per-frame to 0.')
                 return s
             }
-            refreshAnimationCycle(s)
+            // refreshAnimationCycle(s)
             return { ...s, baseMsPerFrame: a.baseMsPerFrame, playbackStartedTimestamp: undefined }
         case SET_REPLAY_RATE:
             // Negative values are allowed and will just play it backwards.
             if (a.newRate === 0) {
                 console.warn(`Attempt to set new replay rate to 0.`)
                 return s
             }
-            refreshAnimationCycle(s)
+            // refreshAnimationCycle(s)
             return { ...s, replayMultiplier: a.newRate, playbackStartedTimestamp: undefined }
         case SKIP:
             return doSkip(s, a)
         case TO_END:
             s.currentFrameIndex = a.backward ? 0 : s.frameData.length - 1
-            refreshAnimationCycle(s)
+            // refreshAnimationCycle(s)
             return {...s, playbackStartedTimestamp: undefined}
         default: {
             throw Error(`Invalid action type for animation state reducer: ${type}`)
         }
     }
 }
 
+// It's not actually clear that this ever did anything useful.
+// TODO: Deprecated. Remove this function (& references to it) after Sep 2022
+// if the widget's behavior remains normal.
```

Comment:
This function doesn't appear to have actually done anything useful, but I think it was also responsible for a race condition when multiple chains of animation-update requests could get triggered on changing the current frame. I've commented it out for now to see if anything breaks that I didn't notice during testing, but it should likely be removed at some point in the near future.